Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -532,36 +571,222 @@ def preprocess_url(url)
       url
     end
 
-    # livecheck should fetch a URL using brewed curl if the formula/cask
+    # livecheck should fetch a URL using brewed curl if the formula/resource/cask
     # contains a `stable`/`url` or `head` URL `using: :homebrew_curl` that
     # shares the same root domain.
-    sig { params(formula_or_cask: T.any(Formula, Cask::Cask), url: String).returns(T::Boolean) }
-    def use_homebrew_curl?(formula_or_cask, url)
+    sig { params(package_or_resource: T.any(Formula, Cask::Cask), url: String).returns(T::Boolean) }
+    def use_homebrew_curl?(package_or_resource, url)
       url_root_domain = Addressable::URI.parse(url)&.domain
       return false if url_root_domain.blank?
 
       # Collect root domains of URLs with `using: :homebrew_curl`
       homebrew_curl_root_domains = []
-      case formula_or_cask
+      case package_or_resource
       when Formula
         [:stable, :head].each do |spec_name|
-          next unless (spec = formula_or_cask.send(spec_name))
+          next unless (spec = package_or_resource.send(spec_name))
           next unless spec.using == :homebrew_curl
 
           domain = Addressable::URI.parse(spec.url)&.domain
           homebrew_curl_root_domains << domain if domain.present?
         end
       when Cask::Cask
-        return false unless formula_or_cask.url.using == :homebrew_curl
+        return false unless package_or_resource.url.using == :homebrew_curl
 
-        domain = Addressable::URI.parse(formula_or_cask.url.to_s)&.domain
+        domain = Addressable::URI.parse(package_or_resource.url.to_s)&.domain
         homebrew_curl_root_domains << domain if domain.present?
       end
 
       homebrew_curl_root_domains.include?(url_root_domain)
     end
 
-    # Identifies the latest version of the formula and returns a Hash containing
+    # Identifies the latest version of the resource in a given Formulae and returns a Hash containing
+    # the version information for a resource. Returns a nil value if a latest version
+    # couldn't be found for a given resource.
+    sig {
+      params(
+        resource:  Resource,
+        json:      T::Boolean,
+        full_name: T::Boolean,
+        verbose:   T::Boolean,
+        debug:     T::Boolean,
+      ).returns(T.nilable(Hash))
+    }
+    def resource_version(
```

Comment:
You do not seem to have accounted for `livecheck` blocks in resources that contain `skip` – you should ensure that this behaviour is supported and identical to what happens with Formulae and Casks.

Reply:
I also noticed that a resource `livecheck` block using `#skip` didn't work when I tested it, so I second this. At least at the moment, I believe we only need to concern ourselves with `skip` `livecheck` blocks for resources but, to achieve this, we will need to make changes to the `Livecheck::SkipConditions` module.

Off the top of my head:

* The `#formula_or_cask_skip` method should be updated to handle resources and it would probably make sense to rename it to `#package_or_resource_skip` (to align with the other changes here).
* We will need a `RESOURCE_CHECKS` array (like the existing ones for formulae and casks) that simply includes `:package_or_resource_skip`.
* `#skip_information`/`#referenced_skip_information`/`#print_skip_information` will need to be updated to handle resources as well.

If you have any questions about this while looking through the `SkipConditions` module, feel free to ask (e.g., on Slack).